controller/api/destination/watcher/workload_watcher.go

```diff
@@ -641,118 +651,26 @@ func (wp *workloadPublisher) updateExternalWorkload(externalWorkload *ext.Extern
 	wp.mu.Lock()
 	defer wp.mu.Unlock()
 
-	// externalWorkload wasn't ready or there was no backing externalWorkload.
-	// check if passed externalWorkload is ready
-	if wp.externalWorkload == nil {
-		if externalWorkload == nil {
-			wp.log.Trace("ExternalWorkload deletion event already consumed - ignore")
-			return
-		}
-
-		if !externalworkload.IsEwReady(externalWorkload) {
-			wp.log.Tracef("ExternalWorkload %s.%s not ready - ignore", externalWorkload.Name, externalWorkload.Namespace)
-			return
-		}
+	wp.addr.ExternalWorkload = externalWorkload
 
-		wp.log.Debugf("ExternalWorkload %s.%s became ready", externalWorkload.Name, externalWorkload.Namespace)
-		wp.externalWorkload = externalWorkload
-		updated := false
-		for _, l := range wp.listeners {
-			addr, err := wp.createAddress()
-			if err != nil {
-				wp.log.Errorf("Error creating address for externalWorkload: %s", err)
-				continue
-			}
-			if err = l.Update(&addr); err != nil {
-				wp.log.Warnf("Error sending update to listener: %s", err)
-				continue
-			}
-			updated = true
-		}
-		if updated {
-			wp.metrics.incUpdates()
-		}
-		return
+	// Fill in ownership.
+	if wp.addr.ExternalWorkload != nil && len(wp.addr.ExternalWorkload.GetOwnerReferences()) == 1 {
+		wp.addr.OwnerKind = wp.addr.ExternalWorkload.GetOwnerReferences()[0].Kind
+		wp.addr.OwnerName = wp.addr.ExternalWorkload.GetOwnerReferences()[0].Name
```

Comment:
TIOLI: you can abstract this ownership filling into separate functions to make things more readable. Here and in the other place where you are doing it.